staging/src/k8s.io/apiserver/pkg/server/options/server_run_options.go

```diff
@@ -337,5 +351,15 @@ func (s *ServerRunOptions) AddUniversalFlags(fs *pflag.FlagSet) {
 		"This option, if set, represents the maximum amount of grace period the apiserver will wait "+
 		"for active watch request(s) to drain during the graceful server shutdown window.")
 
-	utilfeature.DefaultMutableFeatureGate.AddFlag(fs)
+	s.FeatureGate.DeferErrorsToValidation(true)
+	s.FeatureGate.AddFlag(fs)
+	s.EffectiveVersion.AddFlags(fs, "")
+}
+
+// Complete sets the final emulation version for the feature gate.
+func (s *ServerRunOptions) Complete() error {
+	if s.FeatureGate.Enabled(genericfeatures.EmulationVersion) {
```

Comment:
to be determined: is it weird to have a special-case feature-gate for this feature that controls setting of feature gates based on version progression? an alternative "safe-no-op-by-default" option might be to only call this when the `--emulate-version` flag is explicitly set